bin/node/runtime/src/lib.rs

```diff
@@ -330,6 +330,12 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl frame_system_rpc_runtime_api::AccountNonceApi<Block, AccountId, Index> for Runtime {
+		fn account_nonce(account: AccountId) -> Index {
+			System::account_nonce(account)
```

Comment:
This will not take the transaction pool into account (where is substrate it does). Why don't use the extension defined in substrate repo that contains this logic?

Another option would be to have relay detect if this method is available (by calling `rpc_methods` first) and falling back to `System::account_nonce` state read instead).

Reply:
> This will not take the transaction pool into account (where is substrate it does). Why don't use the extension defined in substrate repo that contains this logic?

You're looking at runtime code now :) Of course it won't take transaction pool into account. And substrate runtime can't do this either - it can't access pool.

> Another option would be to have relay detect if this method is available (by calling rpc_methods first) and falling back to System::account_nonce state read instead).

Our node provides required RPC - please look above, at the changes to node code (`with_rpc_extensions() { ... FullSystem::new() ... }`)

Reply:
:man_facepalming: Sorry, I was sure that this is RPC implementation, sorry about that!